pax/simulation.py

```diff
@@ -108,30 +109,35 @@ def __init__(self, config_to_init):
         # Load pdf for single photoelectron, if available
         ##
         if c.get('photon_area_distribution'):
-            # Extract the spe pdf from a csv file in form of bin, prob.
-            charge = []
-            p = []
-            with open(utils.data_file_name(c['photon_area_distribution']), "r") as f:
-                for line in f.readlines():
-                    q = float(line.split(",")[0])
-                    prob = float(line.split(",")[1])
-                    # don't allow negative charge, which can occur from noise subtraction
-                    prob = prob if q >= 0 else 0
-                    charge.append(q)
-                    p.append(prob)
-
-            # Create a converter from uniform random numbers to SPE gains
-            # note not really pdf, does not necessarily sum to 1 due to forcing probability of negative charge to be 0
-            spe_bins, spe_pdf = np.array(charge), np.array(p)
-            # change spe_bins to have mean equal to 1
-            mean_spe = (spe_bins * spe_pdf).sum() / spe_pdf.sum()
-            spe_bins = spe_bins/mean_spe
-            # convert pdf -> cdf
-            cdfs = np.cumsum(spe_pdf)/np.sum(spe_pdf)
-            self.uniform_to_pe = interp1d(cdfs, spe_bins)
-
+            # Extract the spe pdf from a csv file into a pandas dataframe
+            spe_shapes = pd.read_csv(utils.data_file_name(c['photon_area_distribution']))
+
+            # Create a converter array from uniform random numbers to SPE gains (one interpolator per channel)
+            # Scale the distributions so that they have an SPE mean of 1 and then calculate the cdf            
+            # We have set the distribution of the off channels to be explicitly 0 as a precaution
+            # as of now these channels are
+                ## 1, 2, 12, 26, 34, 62, 65, 79, 86, 88, 102, 118, 130, 134, 135, 139, 
+                ## 148, 150, 152, 162, 178, 183, 190, 198, 206, 213, 214, 234, 239, 244
```

Comment:
__[flake8]__
* [E266] too many leading '#' for block comment
* [E116] unexpected indentation (comment)

<!-- Comment by [SideCI](https://sideci.com) -->